model/task/task.go

```diff
@@ -1001,6 +1010,24 @@ func (t *Task) markAsHostDispatchedWithFunc(doUpdate func(update bson.M) error,
 	return nil
 }
 
+// newTaskOutput returns a new task output with the most up-to-date
+// configuration for the task run. This function should only be used to set the
+// task output field upon task dispatch.
+func (t *Task) newTaskOutput() *taskoutput.TaskOutput {
+	if t.DisplayOnly {
+		return nil
```

Comment:
I don't think I want to error since it does look like the dispatch functions are also called for display tasks and I want the logic for whether a task has task output to live here, centrally. What if I return a `bool` as well to indicate whether the task has task output?